include/posix/pthread.h

```diff
@@ -359,4 +381,117 @@ int pthread_barrierattr_getpshared(const pthread_barrierattr_t *, int *);
 int pthread_barrierattr_setpshared(pthread_barrierattr_t *, int);
 */
 
+/* Base Pthread related APIs */
+
+/**
+ * @brief POSIX pthread attributes initialization
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_init(pthread_attr_t *attr)
+{
+	*attr = initpthreadattrs;
+	return 0;
+}
+
+/**
+ * @brief POSIX pthread attributes destroy
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_destroy(pthread_attr_t *attr)
+{
+	return 0;
+}
+
+/**
+ * @brief POSIX get pthread scheduling parameter
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_getschedparam(const pthread_attr_t *attr,
+					     struct sched_param *schedparam)
+{
+	schedparam->priority = attr->priority;
+	return 0;
+}
+
+/**
+ * @brief POSIX get pthread stack inforamtion from thread attribute
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_getstack(const pthread_attr_t *attr,
+					void **stackaddr, size_t *stacksize)
+{
+	*stackaddr = attr->stack;
+	*stacksize = attr->stacksize;
+	return 0;
+}
+
+/**
+ * @brief POSIX get stack size from thread attribute
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_attr_getstacksize(const pthread_attr_t *attr,
+					    size_t *stacksize)
+{
+	*stacksize = attr->stacksize;
+	return 0;
+}
+/**
+ * @brief are Both pthreads same
+ *
+ * See IEEE 1003.1
+ */
+static inline int pthread_equal(pthread_t pt1, pthread_t pt2)
+{
+	return (pt1 == pt2);
+}
+
+/**
+ * @brief pthread exits the system.
+ *
+ * See IEEE 1003.1
+ */
+static inline void pthread_exit(void *retval)
```

Comment:
Do we have a `pthread_join()`?

Reply:
As of now we do not have pthread_join(). We need to figure out how to implement it. Before that we will target other POSIX APIs.

Reply:
One way to do this is the following:

```C
struct pthread {
    struct k_thread thread;
    void *retval;
};

typedef pthread_t struct pthread;
```

This way, it's possible to use type-punning to convert between a `struct k_thread` and a `struct pthread`.  And, for pthreads, it's possible to set/obtain the return value.  So, this function could be implemented like so:

```C
static void pthread_exit(void *retval)
{
        pthread_t *self = (pthread_t *)k_current_get();

        self->retval = retval;

        k_thread_abort(self);
}
```

Then, `pthread_join()` will get the `retval` from `pthread_t`.  Calling `pthread_exit()` should also give a mutex that `pthread_join()` is waiting on, before it aborts itself.

Reply:
Yes I got your Idea, but I wanted to enhance it to be compatible with pthread_join after first set of patch merge. 

For making pthread_join() we need to have 2 members (mutex and Condition variable to signal the state exit). But before that i am waiting for my first set of patches to be merged. I am planning to take this as enhancement, 
struct pthread {
    struct k_thread thread;
    void *retval;
     pthread_mutex_t state_lock;   /* Locks the state.  */
     pthread_cond_t state_cond;    /* Signalled when the state changes.  */
};